File: usemap.php
<?php

$arr = [
  [
    'name' => 'Taro',
    'age'  => 24
  ],
  [
    'name' => 'Goro',
    'age'  => 25
  ],
  [
    'name' => 'Jiro',
    'age'  => 32
  ]
];

$result = array_map(function($obj) { return $obj['age']; }, $arr);

print_r($result);
?>

File: usemap.py
hoge = [
  {
    'name': 'Taro',
    'age' : 24,
  },
  {
    'name': 'Goro',
    'age' : 25,
  },
  {
    'name': 'Jiro',
    'age' : 32,
  }
]

age_map = map(lambda x: x['age'], hoge)
print(age_map)

age_comprehension = [p['age'] for p in hoge]
print(age_comprehension)
